src/components/AppConfig/Vector.tsx

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+
+import { Field, FieldSet, Input, Select, Switch, useStyles2 } from "@grafana/ui";
+
+import { testIds } from "components/testIds";
+import { getStyles } from "./AppConfig";
+
+export interface VectorSettings {
+  // Whether the vector service should be enabled.
+  enabled?: boolean;
+  // The model used by the embedder and for embeddings stored in the store.
+  model?: string;
+  // Embedder settings.
+  embed?: EmbedderSettings;
+  // Store settings.
+  store?: StoreSettings;
+};
+
+interface OpenAIEmbedderSettings {
+  url?: string;
```

Comment:
Do we need Azure OpenAI support here as well? 🤔

Reply:
Yeah we probably will do... the backend won't handle it correctly right now though, needs some unification once all this is merged.